docs/rules/prefer-async-await.md

```diff
@@ -0,0 +1,48 @@
+# Prefer `async`/`await` over `.then()` (prefer-async-await)
+
+With the addition of `async`/`await` in ES2017, one can write asynchronous code
+in a synchronous-like manner.
+
+## Rule Details
+
+This rule flags `then` calls so they can be converted to use `async`/`await`.
+
+Examples of **incorrect** code for this rule:
+
+```js
+/*eslint prefer-async-await: "error"*/
+
+getData()
+  .then((data) => console.log(data))
+  .catch((error) => console.error(error))
+  .finally(() => cleanUp())
+
+dataPromise.then(function(data) { }).then(x).catch()
+```
+
+Examples of **correct** code for this rule:
+
+```js
+/*eslint prefer-async-await: "error"*/
+
+try {
+    let data = await getData();
```

Comment:
Can you put this code into an `async function`? It's invalid syntax as-is.

(I'm mentioning this because the process of fixing errors from this rule could easily be someone's first experience using async/await, and they might end up confused about why `await` isn't working.)